### PR TITLE
fix(arm): Fix arm graph breadcrumbs

### PR DESCRIPTION
**Pull request by omriyoffe-panw**
# User description
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    We use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the other types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Each prefix should be accompanied by a scope that specifies the targeted framework. If uncertain, use 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
<p>Refactor the <code>Runner</code> class in <code>checkov/arm/runner.py</code> to improve file path handling and breadcrumb integration. Introduce <code>clean_file_path</code> in <code>checkov/arm/utils.py</code> to sanitize file paths. Enhance breadcrumb functionality by incorporating <code>GraphRecord</code> for better tracking of resources in the ARM graph.</p>
    <table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/bridgecrewio/checkov/6869?tool=ast&topic=File+Path+Handling>File Path Handling</a>
        </td><td>Improve file path handling in the <code>Runner</code> class by using <code>clean_file_path</code> to sanitize paths and <code>extract_file_path_from_abs_path</code> to convert absolute paths to relative paths.<details><summary>Modified files (2)</summary><ul><li>checkov/arm/runner.py</li>
<li>checkov/arm/utils.py</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>taviassaf</td><td>feat-azure-add-new-pol...</td><td>November 17, 2024</td></tr>
<tr><td>omriyoffe-panw</td><td>fix-arm-finish-variabl...</td><td>November 07, 2024</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/bridgecrewio/checkov/6869?tool=ast&topic=Breadcrumb+Integration>Breadcrumb Integration</a>
        </td><td>Enhance breadcrumb functionality in the <code>Runner</code> class by integrating <code>GraphRecord</code> to track resources more effectively.<details><summary>Modified files (1)</summary><ul><li>checkov/arm/runner.py</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>omriyoffe-panw</td><td>fix-arm-finish-variabl...</td><td>November 07, 2024</td></tr>
<tr><td>lirshindalman</td><td>feat-arm-Add-arm-defin...</td><td>October 30, 2024</td></tr></table></details></td></tr></table>This pull request is reviewed by Baz. Join @omriyoffe-panw and the rest of your team on <a href=https://baz.co/login>(Baz)</a>.

### CODE REVIEW

**Review by lirshindalman — APPROVED** [review]

File: checkov/arm/runner.py [hunk 1]
```diff
@@ -304,5 +306,12 @@ def build_record(
...
         )
+        if self.breadcrumbs:
+            breadcrumb = self.breadcrumbs.get(record.file_path, {}).get(record.resource)
+            if breadcrumb:
+                record = GraphRecord(record, breadcrumb)
         record.set_guideline(check.guideline)
         report.add_record(record=record)
+
+    def extract_file_path_from_abs_path(self, path: Path) -> str:
+        return f"/{os.path.relpath(path, self.root_folder)}"
```
Comment (bo156) [inline checkov/arm/runner.py:317]: this will not work on all OS, use `os.path.sep` instead of `/`.

Comment (omriyoffe-panw) [reply checkov/arm/runner.py]: Done

File: checkov/arm/runner.py [hunk 2]
```diff
@@ -138,6 +140,7 @@ def add_python_check_results(self, report: Report, runner_filter: RunnerFilter,
                 path_to_convert = (os.path.join(root_folder, arm_file)) if root_folder else arm_file
 
             file_abs_path = os.path.abspath(path_to_convert)
+            cleaned_path = clean_file_path(Path(arm_file))
```
Comment (bo156) [inline checkov/arm/runner.py:143]: @omriyoffe-panw could you also add a test to check this use-case?

Comment (omriyoffe-panw) [reply checkov/arm/runner.py]: I removed this since the python checks are not relevant for the custom policies preview flow

**Review by bo156 — APPROVED** [review]

**Pull request merged**